SSA/Projects/FullyHomomorphicEncryption/Basic.lean

```diff
@@ -0,0 +1,66 @@
+import Mathlib.RingTheory.Polynomial.Quotient
+import Mathlib.Data.Zmod.Basic
+import SSA.Core.WellTypedFramework
+
+open Polynomial -- for R[X] notation
+
+-- variable {q : ℤ} [Fact (q > 1)] {n : Nat}
+-- def qZ : Ideal ℤ := Ideal.span {q}
+-- def R := (ℤ ⧸ @qZ q) -- is there a way to avoid explicitly passing q here?
+variable (q : Nat) [Fact (q > 1)] (n : Nat)
+def R := ZMod q
```

Comment:
@goens make this an `abbrev` so Lean can see through `R` for typeclass inference. Alternatively, mark the `def` as `@[reducible]` for the same reason.